buffer/src/main/java/io/netty/buffer/PoolChunk.java

```diff
@@ -531,8 +526,8 @@ private long collapseNext(long handle) {
             int runOffset = runOffset(handle);
             int runPages = runPages(handle);
 
-            Long nextRun = getAvailRunByOffset(runOffset + runPages);
-            if (nextRun == null) {
+            long nextRun = getAvailRunByOffset(runOffset + runPages);
+            if (nextRun == 0) {
```

Comment:
```suggestion
            if (nextRun == -1) {
```
== 0 seems like a bug